tests/backup/backup_test.go

```diff
@@ -86,6 +88,508 @@ var _ = Describe("{UserGroupManagement}", func() {
+						resp, err := Inst().Backup.InspectRestore(ctx, restoreInspectRequest)
+						restoreResponseStatus := resp.GetRestore().GetStatus()
+						log.FailOnError(err, "Failed getting restore status for - %s", laterRestoreName)
+
+						if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_PartialSuccess {
+							log.Infof("Restore status - %s", restoreResponseStatus)
+							log.InfoD("Status of %s - [%s]",
+								laterRestoreName, restoreResponseStatus.GetStatus())
+							log.InfoD("Condition fulfilled. Proceeding to confirm Restore 1")
+							return "", false, nil
+						} else if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Success {
+							log.Infof("Restore status - %s", restoreResponseStatus)
+							log.InfoD("Status of %s - [%s]",
+								laterRestoreName, restoreResponseStatus.GetStatus())
+
+							err := fmt.Errorf("This is not supposed to happen. This indicates that Restore 2 didn't face conflicts with 'Cluster-level resources' from Restore 1")
+							log.FailOnError(err, "Unexpected Status")
+							return "", false, err
+						} else if restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Aborted ||
+							restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Failed ||
+							restoreResponseStatus.GetStatus() == api.RestoreInfo_StatusInfo_Deleting {
+							log.Infof("Restore status - %s", restoreResponseStatus)
+							log.InfoD("Status of %s - [%s]",
+								laterRestoreName, restoreResponseStatus.GetStatus())
+
+							err := fmt.Errorf("Something must have gone wrong. You can try restarting the test")
+							log.FailOnError(err, "Unexpected Status")
+							return "", false, err
+						}
+
+						return "", true, fmt.Errorf("waiting for restore status to be PartialSuccess; got [%s]",
+							restoreResponseStatus.GetStatus())
+					}
+					_, err = task.DoRetryWithTimeout(restorePartialSuccessCheck, 10*time.Minute, 30*time.Second)
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Restore 2 (%s) is PartialSuccess", laterRestoreName))
+
+					// Validation of Restore
+					restoreInspectRequest = &api.RestoreInspectRequest{
+						Name:  laterRestoreName,
+						OrgId: orgID,
+					}
+
+					backupDriver := Inst().Backup
+					resp, err := backupDriver.InspectRestore(ctx, restoreInspectRequest)
+					log.FailOnError(err, "Failed to inspect restore: %s", laterRestoreName)
+
+					log.InfoD("Switching to destination context")
+					err = SetDestinationKubeConfig()
+					log.FailOnError(err, "Failed to switch to context to destination cluster")
+
+					rstCtx, err := ValidateRestore(resp, backupContexts[i], namespaceMapping)
+
+					log.InfoD("Switching to default context")
+					err1 := SetClusterContext("")
+					log.FailOnError(err1, "Failed to SetClusterContext to default cluster")
+
+					dash.VerifyFatal(err, nil, fmt.Sprintf("Restore 2 (%s) Validation", laterRestoreName))
+
+					restoreLaterContexts = append(restoreLaterContexts, rstCtx)
```

Comment:
Why are you maintaining the list here? You are not using it anyway.

Reply:
I'm technically supposed to use this list to clear up the restored namespaces, but I'm not doing it here as I can still manage to run this test case successfully multiple times, without clearing stuff.

The problem is that we currently don't clear restored namespaces anywhere, which can be helpful if done automatically. And in my case, the test will end up fatally the next time, if I don't clear stuff in "restore" (though, restore-later" is optional as the aim is to clear cluster-level resources).

The reason why I'm not clearing the "reastore-later" namespace is because one I clear "restore" namespace, the cluster-level resources are deleted and trying to delete them once again in "restore-later" will generate an error.

THE PROBLEM: In `ValidateAndDestory` function, we don't have an option to excuse errors, as it fails it definitely. Until the function is rewritten such that we have the choice of handling/ignoring errors, I cannot clear the "restore-later" automatically. Which is why, I'm leaving it in there, to be taken care of on a later date.